internal/cli/cmd/install.go

```diff
@@ -233,3 +233,85 @@ cilium upgrade --version %s
 
 	return cmd
 }
+
+func newCmdInstallWithHelm() *cobra.Command {
+	var params = install.Parameters{Writer: os.Stdout}
+
+	cmd := &cobra.Command{
+		Use:   "install",
+		Short: "Install Cilium in a Kubernetes cluster using Helm",
+		Long: `Install Cilium in a Kubernetes cluster using Helm
+
+Examples:
+# Install Cilium in current Kubernetes context with default parameters
+cilium install
+
+# Install Cilium into Kubernetes context "kind-cluster1" and also set cluster
+# name and ID to prepare for multi-cluster capabilities.
+cilium install --context kind-cluster1 --helm-set cluster.id=1 --helm-set cluster.name=cluster1
+`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			params.Namespace = namespace
+			installer, err := install.NewK8sInstaller(k8sClient, params)
+			if err != nil {
+				return err
+			}
+			cmd.SilenceUsage = true
+			if err := installer.InstallWithHelm(context.Background(), k8sClient.RESTClientGetter); err != nil {
+				fatalf("Unable to install Cilium: %s", err)
+			}
+			return nil
+		},
+	}
+
+	// We can't get rid of --cluster-name until we fix https://github.com/cilium/cilium-cli/issues/1347.
+	cmd.Flags().StringVar(&params.ClusterName, "cluster-name", "", "Name of the cluster")
+	cmd.Flags().MarkDeprecated("cluster-name", "This can now be overridden via `helm-set` (Helm value: `cluster.name`).")
+	cmd.Flags().StringVar(&params.Version, "version", defaults.Version, "Cilium version to install")
+	cmd.Flags().StringVar(&params.DatapathMode, "datapath-mode", "", "Datapath mode to use { tunnel | aws-eni | gke | azure | aks-byocni } (default: autodetected).")
+	cmd.Flags().StringVar(&params.HelmChartDirectory, "chart-directory", "", "Helm chart directory")
+	cmd.Flags().StringSliceVar(&params.HelmOpts.ValueFiles, "helm-values", []string{}, "Specify helm values in a YAML file or a URL (can specify multiple)")
+	cmd.Flags().StringArrayVar(&params.HelmOpts.Values, "helm-set", []string{}, "Set helm values on the command line (can specify multiple or separate values with commas: key1=val1,key2=val2)")
+	cmd.Flags().StringArrayVar(&params.HelmOpts.StringValues, "helm-set-string", []string{}, "Set helm STRING values on the command line (can specify multiple or separate values with commas: key1=val1,key2=val2)")
+	cmd.Flags().StringArrayVar(&params.HelmOpts.FileValues, "helm-set-file", []string{}, "Set helm values from respective files specified via the command line (can specify multiple or separate values with commas: key1=path1,key2=path2)")
+	cmd.Flags().BoolVar(&params.ListVersions, "list-versions", false, "List all the available versions without actually installing")
+	cmd.Flags().BoolVar(&params.Wait, "wait", false, "Wait for helm install to finish")
```

Comment:
It's unfortunate these flags are duplicated between this function and `newCmdInstall`. Maybe define them in a common func that is called from both and `newCmdInstall` just adds its additional (deprecated) flags on top?